{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting faker\n",
      "  Downloading Faker-18.3.4-py3-none-any.whl (1.7 MB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.7/1.7 MB\u001b[0m \u001b[31m5.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0ma \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: pandas in /home/barteus/Work/tutorials/kubecon2023demo/.venv/lib/python3.10/site-packages (2.0.0)\n",
      "Requirement already satisfied: python-dateutil>=2.4 in /home/barteus/Work/tutorials/kubecon2023demo/.venv/lib/python3.10/site-packages (from faker) (2.8.2)\n",
      "Requirement already satisfied: tzdata>=2022.1 in /home/barteus/Work/tutorials/kubecon2023demo/.venv/lib/python3.10/site-packages (from pandas) (2023.3)\n",
      "Requirement already satisfied: numpy>=1.21.0 in /home/barteus/Work/tutorials/kubecon2023demo/.venv/lib/python3.10/site-packages (from pandas) (1.24.2)\n",
      "Requirement already satisfied: pytz>=2020.1 in /home/barteus/Work/tutorials/kubecon2023demo/.venv/lib/python3.10/site-packages (from pandas) (2023.3)\n",
      "Requirement already satisfied: six>=1.5 in /home/barteus/Work/tutorials/kubecon2023demo/.venv/lib/python3.10/site-packages (from python-dateutil>=2.4->faker) (1.16.0)\n",
      "Installing collected packages: faker\n",
      "Successfully installed faker-18.3.4\n"
     ]
    }
   ],
   "source": [
    "%pip install faker pandas -q"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import re\n",
    "\n",
    "import faker\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def transform_columns(columns):\n",
    "    return [re.sub(\"\\(|\\)\", \"\", re.sub(\"-|--|\\.|\\s\", \"_\", c.lower())).replace(\"__\", \"_\")\n",
    "            for c in columns]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": false
   },
   "source": [
    "# Clinical Data Breast Cancer dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Complete TCGA ID</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age at Initial Pathologic Diagnosis</th>\n",
       "      <th>ER Status</th>\n",
       "      <th>PR Status</th>\n",
       "      <th>HER2 Final Status</th>\n",
       "      <th>Tumor</th>\n",
       "      <th>Tumor--T1 Coded</th>\n",
       "      <th>Node</th>\n",
       "      <th>Node-Coded</th>\n",
       "      <th>...</th>\n",
       "      <th>PAM50 mRNA</th>\n",
       "      <th>SigClust Unsupervised mRNA</th>\n",
       "      <th>SigClust Intrinsic mRNA</th>\n",
       "      <th>miRNA Clusters</th>\n",
       "      <th>methylation Clusters</th>\n",
       "      <th>RPPA Clusters</th>\n",
       "      <th>CN Clusters</th>\n",
       "      <th>Integrated Clusters (with PAM50)</th>\n",
       "      <th>Integrated Clusters (no exp)</th>\n",
       "      <th>Integrated Clusters (unsup exp)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-A2-A0T2</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>66</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T3</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N3</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>0</td>\n",
       "      <td>-13</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-A2-A0CM</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>40</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>-12</td>\n",
       "      <td>-13</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>Basal</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-BH-A18V</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>48</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N1</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>-12</td>\n",
       "      <td>-13</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-BH-A18Q</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>56</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N1</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>-12</td>\n",
       "      <td>-13</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-BH-A0E0</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>38</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T3</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N3</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>0</td>\n",
       "      <td>-13</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Complete TCGA ID  Gender  Age at Initial Pathologic Diagnosis ER Status   \n",
       "0     TCGA-A2-A0T2  FEMALE                                   66  Negative  \\\n",
       "1     TCGA-A2-A0CM  FEMALE                                   40  Negative   \n",
       "2     TCGA-BH-A18V  FEMALE                                   48  Negative   \n",
       "3     TCGA-BH-A18Q  FEMALE                                   56  Negative   \n",
       "4     TCGA-BH-A0E0  FEMALE                                   38  Negative   \n",
       "\n",
       "  PR Status HER2 Final Status Tumor Tumor--T1 Coded Node Node-Coded  ...   \n",
       "0  Negative          Negative    T3         T_Other   N3   Positive  ...  \\\n",
       "1  Negative          Negative    T2         T_Other   N0   Negative  ...   \n",
       "2  Negative          Negative    T2         T_Other   N1   Positive  ...   \n",
       "3  Negative          Negative    T2         T_Other   N1   Positive  ...   \n",
       "4  Negative          Negative    T3         T_Other   N3   Positive  ...   \n",
       "\n",
       "   PAM50 mRNA SigClust Unsupervised mRNA SigClust Intrinsic mRNA   \n",
       "0  Basal-like                          0                     -13  \\\n",
       "1  Basal-like                        -12                     -13   \n",
       "2  Basal-like                        -12                     -13   \n",
       "3  Basal-like                        -12                     -13   \n",
       "4  Basal-like                          0                     -13   \n",
       "\n",
       "  miRNA Clusters methylation Clusters RPPA Clusters  CN Clusters   \n",
       "0              3                    5         Basal            3  \\\n",
       "1              4                    4         Basal            4   \n",
       "2              5                    5         Basal            1   \n",
       "3              5                    5         Basal            1   \n",
       "4              5                    5         Basal            1   \n",
       "\n",
       "   Integrated Clusters (with PAM50)  Integrated Clusters (no exp)   \n",
       "0                                 2                             2  \\\n",
       "1                                 2                             1   \n",
       "2                                 2                             2   \n",
       "3                                 2                             2   \n",
       "4                                 2                             2   \n",
       "\n",
       "   Integrated Clusters (unsup exp)  \n",
       "0                                2  \n",
       "1                                1  \n",
       "2                                2  \n",
       "3                                2  \n",
       "4                                2  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_breast_cancer = pd.read_csv(\"./app/data/clinical_data_breast_cancer.csv\")\n",
    "df_breast_cancer.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 105 entries, 0 to 104\n",
      "Data columns (total 30 columns):\n",
      " #   Column                               Non-Null Count  Dtype  \n",
      "---  ------                               --------------  -----  \n",
      " 0   Complete TCGA ID                     105 non-null    object \n",
      " 1   Gender                               105 non-null    object \n",
      " 2   Age at Initial Pathologic Diagnosis  105 non-null    int64  \n",
      " 3   ER Status                            105 non-null    object \n",
      " 4   PR Status                            105 non-null    object \n",
      " 5   HER2 Final Status                    105 non-null    object \n",
      " 6   Tumor                                105 non-null    object \n",
      " 7   Tumor--T1 Coded                      105 non-null    object \n",
      " 8   Node                                 105 non-null    object \n",
      " 9   Node-Coded                           105 non-null    object \n",
      " 10  Metastasis                           105 non-null    object \n",
      " 11  Metastasis-Coded                     105 non-null    object \n",
      " 12  AJCC Stage                           105 non-null    object \n",
      " 13  Converted Stage                      105 non-null    object \n",
      " 14  Survival Data Form                   105 non-null    object \n",
      " 15  Vital Status                         105 non-null    object \n",
      " 16  Days to Date of Last Contact         105 non-null    int64  \n",
      " 17  Days to date of Death                11 non-null     float64\n",
      " 18  OS event                             105 non-null    int64  \n",
      " 19  OS Time                              105 non-null    int64  \n",
      " 20  PAM50 mRNA                           105 non-null    object \n",
      " 21  SigClust Unsupervised mRNA           105 non-null    int64  \n",
      " 22  SigClust Intrinsic mRNA              105 non-null    int64  \n",
      " 23  miRNA Clusters                       105 non-null    int64  \n",
      " 24  methylation Clusters                 105 non-null    int64  \n",
      " 25  RPPA Clusters                        105 non-null    object \n",
      " 26  CN Clusters                          105 non-null    int64  \n",
      " 27  Integrated Clusters (with PAM50)     105 non-null    int64  \n",
      " 28  Integrated Clusters (no exp)         105 non-null    int64  \n",
      " 29  Integrated Clusters (unsup exp)      105 non-null    int64  \n",
      "dtypes: float64(1), int64(12), object(17)\n",
      "memory usage: 24.7+ KB\n"
     ]
    }
   ],
   "source": [
    "df_breast_cancer.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "fake = faker.Faker()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>complete_tcga_id</th>\n",
       "      <th>gender</th>\n",
       "      <th>age_at_initial_pathologic_diagnosis</th>\n",
       "      <th>er_status</th>\n",
       "      <th>pr_status</th>\n",
       "      <th>her2_final_status</th>\n",
       "      <th>tumor</th>\n",
       "      <th>tumor_t1_coded</th>\n",
       "      <th>node</th>\n",
       "      <th>node_coded</th>\n",
       "      <th>...</th>\n",
       "      <th>sigclust_intrinsic_mrna</th>\n",
       "      <th>mirna_clusters</th>\n",
       "      <th>methylation_clusters</th>\n",
       "      <th>rppa_clusters</th>\n",
       "      <th>cn_clusters</th>\n",
       "      <th>integrated_clusters_with_pam50</th>\n",
       "      <th>integrated_clusters_no_exp</th>\n",
       "      <th>integrated_clusters_unsup_exp</th>\n",
       "      <th>first_name</th>\n",
       "      <th>last_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-A2-A0T2</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>66</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T3</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N3</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>-13</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>Melissa</td>\n",
       "      <td>Johnson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-A2-A0CM</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>40</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>-13</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>Basal</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Cynthia</td>\n",
       "      <td>Watson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-BH-A18V</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>48</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N1</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>-13</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>Tracy</td>\n",
       "      <td>Peters</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-BH-A18Q</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>56</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N1</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>-13</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>Angela</td>\n",
       "      <td>Baxter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-BH-A0E0</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>38</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T3</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N3</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>-13</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>Christina</td>\n",
       "      <td>Cohen</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  complete_tcga_id  gender  age_at_initial_pathologic_diagnosis er_status   \n",
       "0     TCGA-A2-A0T2  FEMALE                                   66  Negative  \\\n",
       "1     TCGA-A2-A0CM  FEMALE                                   40  Negative   \n",
       "2     TCGA-BH-A18V  FEMALE                                   48  Negative   \n",
       "3     TCGA-BH-A18Q  FEMALE                                   56  Negative   \n",
       "4     TCGA-BH-A0E0  FEMALE                                   38  Negative   \n",
       "\n",
       "  pr_status her2_final_status tumor tumor_t1_coded node node_coded  ...   \n",
       "0  Negative          Negative    T3        T_Other   N3   Positive  ...  \\\n",
       "1  Negative          Negative    T2        T_Other   N0   Negative  ...   \n",
       "2  Negative          Negative    T2        T_Other   N1   Positive  ...   \n",
       "3  Negative          Negative    T2        T_Other   N1   Positive  ...   \n",
       "4  Negative          Negative    T3        T_Other   N3   Positive  ...   \n",
       "\n",
       "  sigclust_intrinsic_mrna mirna_clusters methylation_clusters rppa_clusters   \n",
       "0                     -13              3                    5         Basal  \\\n",
       "1                     -13              4                    4         Basal   \n",
       "2                     -13              5                    5         Basal   \n",
       "3                     -13              5                    5         Basal   \n",
       "4                     -13              5                    5         Basal   \n",
       "\n",
       "  cn_clusters integrated_clusters_with_pam50  integrated_clusters_no_exp   \n",
       "0           3                              2                           2  \\\n",
       "1           4                              2                           1   \n",
       "2           1                              2                           2   \n",
       "3           1                              2                           2   \n",
       "4           1                              2                           2   \n",
       "\n",
       "   integrated_clusters_unsup_exp  first_name  last_name  \n",
       "0                              2     Melissa    Johnson  \n",
       "1                              1     Cynthia     Watson  \n",
       "2                              2       Tracy     Peters  \n",
       "3                              2      Angela     Baxter  \n",
       "4                              2   Christina      Cohen  \n",
       "\n",
       "[5 rows x 32 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_breast_cancer[\"First Name\"] = [fake.first_name_female() for _ in\n",
    "                                  range(df_breast_cancer.shape[0])]\n",
    "df_breast_cancer[\"Last Name\"] = [fake.last_name_female() for _ in range(df_breast_cancer.shape[0])]\n",
    "df_breast_cancer.columns = transform_columns(df_breast_cancer.columns)\n",
    "df_breast_cancer.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "df_breast_cancer.to_csv(\"./app/data/clinical_data_breast_cancer_with_columns.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>complete_tcga_id</th>\n",
       "      <th>gender</th>\n",
       "      <th>age_at_initial_pathologic_diagnosis</th>\n",
       "      <th>er_status</th>\n",
       "      <th>pr_status</th>\n",
       "      <th>her2_final_status</th>\n",
       "      <th>tumor</th>\n",
       "      <th>tumor_t1_coded</th>\n",
       "      <th>node</th>\n",
       "      <th>node_coded</th>\n",
       "      <th>...</th>\n",
       "      <th>sigclust_intrinsic_mrna</th>\n",
       "      <th>mirna_clusters</th>\n",
       "      <th>methylation_clusters</th>\n",
       "      <th>rppa_clusters</th>\n",
       "      <th>cn_clusters</th>\n",
       "      <th>integrated_clusters_with_pam50</th>\n",
       "      <th>integrated_clusters_no_exp</th>\n",
       "      <th>integrated_clusters_unsup_exp</th>\n",
       "      <th>first_name</th>\n",
       "      <th>last_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-A2-A0T2</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>66</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T3</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N3</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>-13</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>Melissa</td>\n",
       "      <td>Johnson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-A2-A0CM</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>40</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>-13</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>Basal</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Cynthia</td>\n",
       "      <td>Watson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-BH-A18V</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>48</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N1</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>-13</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>Tracy</td>\n",
       "      <td>Peters</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-BH-A18Q</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>56</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N1</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>-13</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>Angela</td>\n",
       "      <td>Baxter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-BH-A0E0</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>38</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T3</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N3</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>-13</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>Christina</td>\n",
       "      <td>Cohen</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  complete_tcga_id  gender  age_at_initial_pathologic_diagnosis er_status   \n",
       "0     TCGA-A2-A0T2  FEMALE                                   66  Negative  \\\n",
       "1     TCGA-A2-A0CM  FEMALE                                   40  Negative   \n",
       "2     TCGA-BH-A18V  FEMALE                                   48  Negative   \n",
       "3     TCGA-BH-A18Q  FEMALE                                   56  Negative   \n",
       "4     TCGA-BH-A0E0  FEMALE                                   38  Negative   \n",
       "\n",
       "  pr_status her2_final_status tumor tumor_t1_coded node node_coded  ...   \n",
       "0  Negative          Negative    T3        T_Other   N3   Positive  ...  \\\n",
       "1  Negative          Negative    T2        T_Other   N0   Negative  ...   \n",
       "2  Negative          Negative    T2        T_Other   N1   Positive  ...   \n",
       "3  Negative          Negative    T2        T_Other   N1   Positive  ...   \n",
       "4  Negative          Negative    T3        T_Other   N3   Positive  ...   \n",
       "\n",
       "  sigclust_intrinsic_mrna mirna_clusters methylation_clusters rppa_clusters   \n",
       "0                     -13              3                    5         Basal  \\\n",
       "1                     -13              4                    4         Basal   \n",
       "2                     -13              5                    5         Basal   \n",
       "3                     -13              5                    5         Basal   \n",
       "4                     -13              5                    5         Basal   \n",
       "\n",
       "  cn_clusters integrated_clusters_with_pam50  integrated_clusters_no_exp   \n",
       "0           3                              2                           2  \\\n",
       "1           4                              2                           1   \n",
       "2           1                              2                           2   \n",
       "3           1                              2                           2   \n",
       "4           1                              2                           2   \n",
       "\n",
       "   integrated_clusters_unsup_exp  first_name  last_name  \n",
       "0                              2     Melissa    Johnson  \n",
       "1                              1     Cynthia     Watson  \n",
       "2                              2       Tracy     Peters  \n",
       "3                              2      Angela     Baxter  \n",
       "4                              2   Christina      Cohen  \n",
       "\n",
       "[5 rows x 32 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "read_csv = pd.read_csv(\"./app/data/clinical_data_breast_cancer_with_columns.csv\")\n",
    "read_csv.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": false
   },
   "source": [
    "# PAM50 proteins dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>RefSeqProteinID</th>\n",
       "      <th>Species</th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIA</td>\n",
       "      <td>NP_006524</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>melanoma inhibitory activity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FGFR4</td>\n",
       "      <td>NP_002002</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>fibroblast growth factor receptor 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FGFR4</td>\n",
       "      <td>NP_998812</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>fibroblast growth factor receptor 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FGFR4</td>\n",
       "      <td>NP_075252</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>fibroblast growth factor receptor 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GPR160</td>\n",
       "      <td>NP_055188</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>G protein-coupled receptor 160</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  GeneSymbol RefSeqProteinID       Species   \n",
       "0        MIA       NP_006524  Homo sapiens  \\\n",
       "1      FGFR4       NP_002002  Homo sapiens   \n",
       "2      FGFR4       NP_998812  Homo sapiens   \n",
       "3      FGFR4       NP_075252  Homo sapiens   \n",
       "4     GPR160       NP_055188  Homo sapiens   \n",
       "\n",
       "                             Gene Name  \n",
       "0         melanoma inhibitory activity  \n",
       "1  fibroblast growth factor receptor 4  \n",
       "2  fibroblast growth factor receptor 4  \n",
       "3  fibroblast growth factor receptor 4  \n",
       "4       G protein-coupled receptor 160  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_pam50 =pd.read_csv(\"./app/data/PAM50_proteins.csv\")\n",
    "df_pam50.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genesymbol</th>\n",
       "      <th>refseqproteinid</th>\n",
       "      <th>species</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIA</td>\n",
       "      <td>NP_006524</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>melanoma inhibitory activity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FGFR4</td>\n",
       "      <td>NP_002002</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>fibroblast growth factor receptor 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FGFR4</td>\n",
       "      <td>NP_998812</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>fibroblast growth factor receptor 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FGFR4</td>\n",
       "      <td>NP_075252</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>fibroblast growth factor receptor 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GPR160</td>\n",
       "      <td>NP_055188</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>G protein-coupled receptor 160</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  genesymbol refseqproteinid       species   \n",
       "0        MIA       NP_006524  Homo sapiens  \\\n",
       "1      FGFR4       NP_002002  Homo sapiens   \n",
       "2      FGFR4       NP_998812  Homo sapiens   \n",
       "3      FGFR4       NP_075252  Homo sapiens   \n",
       "4     GPR160       NP_055188  Homo sapiens   \n",
       "\n",
       "                             gene_name  \n",
       "0         melanoma inhibitory activity  \n",
       "1  fibroblast growth factor receptor 4  \n",
       "2  fibroblast growth factor receptor 4  \n",
       "3  fibroblast growth factor receptor 4  \n",
       "4       G protein-coupled receptor 160  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_pam50.columns = transform_columns(df_pam50.columns)\n",
    "df_pam50.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genesymbol</th>\n",
       "      <th>refseqproteinid</th>\n",
       "      <th>species</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIA</td>\n",
       "      <td>NP_006524</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>melanoma inhibitory activity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FGFR4</td>\n",
       "      <td>NP_002002</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>fibroblast growth factor receptor 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FGFR4</td>\n",
       "      <td>NP_998812</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>fibroblast growth factor receptor 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FGFR4</td>\n",
       "      <td>NP_075252</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>fibroblast growth factor receptor 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GPR160</td>\n",
       "      <td>NP_055188</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>G protein-coupled receptor 160</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  genesymbol refseqproteinid       species   \n",
       "0        MIA       NP_006524  Homo sapiens  \\\n",
       "1      FGFR4       NP_002002  Homo sapiens   \n",
       "2      FGFR4       NP_998812  Homo sapiens   \n",
       "3      FGFR4       NP_075252  Homo sapiens   \n",
       "4     GPR160       NP_055188  Homo sapiens   \n",
       "\n",
       "                             gene_name  \n",
       "0         melanoma inhibitory activity  \n",
       "1  fibroblast growth factor receptor 4  \n",
       "2  fibroblast growth factor receptor 4  \n",
       "3  fibroblast growth factor receptor 4  \n",
       "4       G protein-coupled receptor 160  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_pam50.to_csv(\"./app/data/PAM50_proteins_with_columns.csv\", index=False)\n",
    "read_df_pam50 = pd.read_csv(\"./app/data/PAM50_proteins_with_columns.csv\")\n",
    "read_df_pam50.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": false
   },
   "source": [
    "# 77 Cancer Proteomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>refseq_accession_number</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>ao_a12d_01tcga</th>\n",
       "      <th>c8_a131_01tcga</th>\n",
       "      <th>ao_a12b_01tcga</th>\n",
       "      <th>bh_a18q_02tcga</th>\n",
       "      <th>c8_a130_02tcga</th>\n",
       "      <th>c8_a138_03tcga</th>\n",
       "      <th>e2_a154_03tcga</th>\n",
       "      <th>...</th>\n",
       "      <th>ao_a12b_34tcga</th>\n",
       "      <th>a2_a0sw_35tcga</th>\n",
       "      <th>ao_a0jl_35tcga</th>\n",
       "      <th>bh_a0bv_35tcga</th>\n",
       "      <th>a2_a0ym_36tcga</th>\n",
       "      <th>bh_a0c7_36tcga</th>\n",
       "      <th>a2_a0sx_36tcga</th>\n",
       "      <th>263d3f_i_cptac</th>\n",
       "      <th>blcdb9_i_cptac</th>\n",
       "      <th>c4155b_c_cptac</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NP_958782</td>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin isoform 1</td>\n",
       "      <td>1.096131</td>\n",
       "      <td>2.609943</td>\n",
       "      <td>-0.659828</td>\n",
       "      <td>0.195341</td>\n",
       "      <td>-0.494060</td>\n",
       "      <td>2.765081</td>\n",
       "      <td>0.862659</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.963904</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.065838</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.398560</td>\n",
       "      <td>0.598585</td>\n",
       "      <td>-0.191285</td>\n",
       "      <td>0.566975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NP_958785</td>\n",
       "      <td>NaN</td>\n",
       "      <td>plectin isoform 1g</td>\n",
       "      <td>1.111370</td>\n",
       "      <td>2.650422</td>\n",
       "      <td>-0.648742</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.503899</td>\n",
       "      <td>2.779709</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.938210</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.055893</td>\n",
       "      <td>0.658143</td>\n",
       "      <td>-0.547749</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.606697</td>\n",
       "      <td>-0.183918</td>\n",
       "      <td>0.578702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NP_958786</td>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin isoform 1a</td>\n",
       "      <td>1.111370</td>\n",
       "      <td>2.650422</td>\n",
       "      <td>-0.654285</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.500619</td>\n",
       "      <td>2.779709</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.943919</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.065838</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.603993</td>\n",
       "      <td>-0.186022</td>\n",
       "      <td>0.576747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NP_000436</td>\n",
       "      <td>NaN</td>\n",
       "      <td>plectin isoform 1c</td>\n",
       "      <td>1.107561</td>\n",
       "      <td>2.646374</td>\n",
       "      <td>-0.632113</td>\n",
       "      <td>0.205377</td>\n",
       "      <td>-0.510459</td>\n",
       "      <td>2.797995</td>\n",
       "      <td>0.866423</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.935355</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.055893</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.603993</td>\n",
       "      <td>-0.186022</td>\n",
       "      <td>0.576747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NP_958781</td>\n",
       "      <td>NaN</td>\n",
       "      <td>plectin isoform 1e</td>\n",
       "      <td>1.115180</td>\n",
       "      <td>2.646374</td>\n",
       "      <td>-0.640428</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.503899</td>\n",
       "      <td>2.787023</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.935355</td>\n",
       "      <td>-0.503853</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.062523</td>\n",
       "      <td>0.651264</td>\n",
       "      <td>-0.556675</td>\n",
       "      <td>-0.395581</td>\n",
       "      <td>0.603993</td>\n",
       "      <td>-0.167079</td>\n",
       "      <td>0.576747</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 86 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  refseq_accession_number gene_symbol           gene_name  ao_a12d_01tcga   \n",
       "0               NP_958782        PLEC   plectin isoform 1        1.096131  \\\n",
       "1               NP_958785         NaN  plectin isoform 1g        1.111370   \n",
       "2               NP_958786        PLEC  plectin isoform 1a        1.111370   \n",
       "3               NP_000436         NaN  plectin isoform 1c        1.107561   \n",
       "4               NP_958781         NaN  plectin isoform 1e        1.115180   \n",
       "\n",
       "   c8_a131_01tcga  ao_a12b_01tcga  bh_a18q_02tcga  c8_a130_02tcga   \n",
       "0        2.609943       -0.659828        0.195341       -0.494060  \\\n",
       "1        2.650422       -0.648742        0.215413       -0.503899   \n",
       "2        2.650422       -0.654285        0.215413       -0.500619   \n",
       "3        2.646374       -0.632113        0.205377       -0.510459   \n",
       "4        2.646374       -0.640428        0.215413       -0.503899   \n",
       "\n",
       "   c8_a138_03tcga  e2_a154_03tcga  ...  ao_a12b_34tcga  a2_a0sw_35tcga   \n",
       "0        2.765081        0.862659  ...       -0.963904       -0.487772  \\\n",
       "1        2.779709        0.870186  ...       -0.938210       -0.487772   \n",
       "2        2.779709        0.870186  ...       -0.943919       -0.487772   \n",
       "3        2.797995        0.866423  ...       -0.935355       -0.487772   \n",
       "4        2.787023        0.870186  ...       -0.935355       -0.503853   \n",
       "\n",
       "   ao_a0jl_35tcga  bh_a0bv_35tcga  a2_a0ym_36tcga  bh_a0c7_36tcga   \n",
       "0        -0.10668       -0.065838        0.655850       -0.552212  \\\n",
       "1        -0.10668       -0.055893        0.658143       -0.547749   \n",
       "2        -0.10668       -0.065838        0.655850       -0.552212   \n",
       "3        -0.10668       -0.055893        0.655850       -0.552212   \n",
       "4        -0.10668       -0.062523        0.651264       -0.556675   \n",
       "\n",
       "   a2_a0sx_36tcga  263d3f_i_cptac  blcdb9_i_cptac  c4155b_c_cptac  \n",
       "0       -0.398560        0.598585       -0.191285        0.566975  \n",
       "1       -0.392601        0.606697       -0.183918        0.578702  \n",
       "2       -0.392601        0.603993       -0.186022        0.576747  \n",
       "3       -0.392601        0.603993       -0.186022        0.576747  \n",
       "4       -0.395581        0.603993       -0.167079        0.576747  \n",
       "\n",
       "[5 rows x 86 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_77proteomes =pd.read_csv(\"./app/data/77_cancer_proteomes_CPTAC_itraq.csv\")\n",
    "df_77proteomes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>refseq_accession_number</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>ao_a12d_01tcga</th>\n",
       "      <th>c8_a131_01tcga</th>\n",
       "      <th>ao_a12b_01tcga</th>\n",
       "      <th>bh_a18q_02tcga</th>\n",
       "      <th>c8_a130_02tcga</th>\n",
       "      <th>c8_a138_03tcga</th>\n",
       "      <th>e2_a154_03tcga</th>\n",
       "      <th>...</th>\n",
       "      <th>ao_a12b_34tcga</th>\n",
       "      <th>a2_a0sw_35tcga</th>\n",
       "      <th>ao_a0jl_35tcga</th>\n",
       "      <th>bh_a0bv_35tcga</th>\n",
       "      <th>a2_a0ym_36tcga</th>\n",
       "      <th>bh_a0c7_36tcga</th>\n",
       "      <th>a2_a0sx_36tcga</th>\n",
       "      <th>263d3f_i_cptac</th>\n",
       "      <th>blcdb9_i_cptac</th>\n",
       "      <th>c4155b_c_cptac</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NP_958782</td>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin isoform 1</td>\n",
       "      <td>1.096131</td>\n",
       "      <td>2.609943</td>\n",
       "      <td>-0.659828</td>\n",
       "      <td>0.195341</td>\n",
       "      <td>-0.494060</td>\n",
       "      <td>2.765081</td>\n",
       "      <td>0.862659</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.963904</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.065838</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.398560</td>\n",
       "      <td>0.598585</td>\n",
       "      <td>-0.191285</td>\n",
       "      <td>0.566975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NP_958785</td>\n",
       "      <td>NaN</td>\n",
       "      <td>plectin isoform 1g</td>\n",
       "      <td>1.111370</td>\n",
       "      <td>2.650422</td>\n",
       "      <td>-0.648742</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.503899</td>\n",
       "      <td>2.779709</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.938210</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.055893</td>\n",
       "      <td>0.658143</td>\n",
       "      <td>-0.547749</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.606697</td>\n",
       "      <td>-0.183918</td>\n",
       "      <td>0.578702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NP_958786</td>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin isoform 1a</td>\n",
       "      <td>1.111370</td>\n",
       "      <td>2.650422</td>\n",
       "      <td>-0.654285</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.500619</td>\n",
       "      <td>2.779709</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.943919</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.065838</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.603993</td>\n",
       "      <td>-0.186022</td>\n",
       "      <td>0.576747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NP_000436</td>\n",
       "      <td>NaN</td>\n",
       "      <td>plectin isoform 1c</td>\n",
       "      <td>1.107561</td>\n",
       "      <td>2.646374</td>\n",
       "      <td>-0.632113</td>\n",
       "      <td>0.205377</td>\n",
       "      <td>-0.510459</td>\n",
       "      <td>2.797995</td>\n",
       "      <td>0.866423</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.935355</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.055893</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.603993</td>\n",
       "      <td>-0.186022</td>\n",
       "      <td>0.576747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NP_958781</td>\n",
       "      <td>NaN</td>\n",
       "      <td>plectin isoform 1e</td>\n",
       "      <td>1.115180</td>\n",
       "      <td>2.646374</td>\n",
       "      <td>-0.640428</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.503899</td>\n",
       "      <td>2.787023</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.935355</td>\n",
       "      <td>-0.503853</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.062523</td>\n",
       "      <td>0.651264</td>\n",
       "      <td>-0.556675</td>\n",
       "      <td>-0.395581</td>\n",
       "      <td>0.603993</td>\n",
       "      <td>-0.167079</td>\n",
       "      <td>0.576747</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 86 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  refseq_accession_number gene_symbol           gene_name  ao_a12d_01tcga   \n",
       "0               NP_958782        PLEC   plectin isoform 1        1.096131  \\\n",
       "1               NP_958785         NaN  plectin isoform 1g        1.111370   \n",
       "2               NP_958786        PLEC  plectin isoform 1a        1.111370   \n",
       "3               NP_000436         NaN  plectin isoform 1c        1.107561   \n",
       "4               NP_958781         NaN  plectin isoform 1e        1.115180   \n",
       "\n",
       "   c8_a131_01tcga  ao_a12b_01tcga  bh_a18q_02tcga  c8_a130_02tcga   \n",
       "0        2.609943       -0.659828        0.195341       -0.494060  \\\n",
       "1        2.650422       -0.648742        0.215413       -0.503899   \n",
       "2        2.650422       -0.654285        0.215413       -0.500619   \n",
       "3        2.646374       -0.632113        0.205377       -0.510459   \n",
       "4        2.646374       -0.640428        0.215413       -0.503899   \n",
       "\n",
       "   c8_a138_03tcga  e2_a154_03tcga  ...  ao_a12b_34tcga  a2_a0sw_35tcga   \n",
       "0        2.765081        0.862659  ...       -0.963904       -0.487772  \\\n",
       "1        2.779709        0.870186  ...       -0.938210       -0.487772   \n",
       "2        2.779709        0.870186  ...       -0.943919       -0.487772   \n",
       "3        2.797995        0.866423  ...       -0.935355       -0.487772   \n",
       "4        2.787023        0.870186  ...       -0.935355       -0.503853   \n",
       "\n",
       "   ao_a0jl_35tcga  bh_a0bv_35tcga  a2_a0ym_36tcga  bh_a0c7_36tcga   \n",
       "0        -0.10668       -0.065838        0.655850       -0.552212  \\\n",
       "1        -0.10668       -0.055893        0.658143       -0.547749   \n",
       "2        -0.10668       -0.065838        0.655850       -0.552212   \n",
       "3        -0.10668       -0.055893        0.655850       -0.552212   \n",
       "4        -0.10668       -0.062523        0.651264       -0.556675   \n",
       "\n",
       "   a2_a0sx_36tcga  263d3f_i_cptac  blcdb9_i_cptac  c4155b_c_cptac  \n",
       "0       -0.398560        0.598585       -0.191285        0.566975  \n",
       "1       -0.392601        0.606697       -0.183918        0.578702  \n",
       "2       -0.392601        0.603993       -0.186022        0.576747  \n",
       "3       -0.392601        0.603993       -0.186022        0.576747  \n",
       "4       -0.395581        0.603993       -0.167079        0.576747  \n",
       "\n",
       "[5 rows x 86 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_77proteomes.columns = transform_columns(df_77proteomes.columns)\n",
    "df_77proteomes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>refseq_accession_number</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>ao_a12d_01tcga</th>\n",
       "      <th>c8_a131_01tcga</th>\n",
       "      <th>ao_a12b_01tcga</th>\n",
       "      <th>bh_a18q_02tcga</th>\n",
       "      <th>c8_a130_02tcga</th>\n",
       "      <th>c8_a138_03tcga</th>\n",
       "      <th>e2_a154_03tcga</th>\n",
       "      <th>...</th>\n",
       "      <th>ao_a12b_34tcga</th>\n",
       "      <th>a2_a0sw_35tcga</th>\n",
       "      <th>ao_a0jl_35tcga</th>\n",
       "      <th>bh_a0bv_35tcga</th>\n",
       "      <th>a2_a0ym_36tcga</th>\n",
       "      <th>bh_a0c7_36tcga</th>\n",
       "      <th>a2_a0sx_36tcga</th>\n",
       "      <th>263d3f_i_cptac</th>\n",
       "      <th>blcdb9_i_cptac</th>\n",
       "      <th>c4155b_c_cptac</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NP_958782</td>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin isoform 1</td>\n",
       "      <td>1.096131</td>\n",
       "      <td>2.609943</td>\n",
       "      <td>-0.659828</td>\n",
       "      <td>0.195341</td>\n",
       "      <td>-0.494060</td>\n",
       "      <td>2.765081</td>\n",
       "      <td>0.862659</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.963904</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.065838</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.398560</td>\n",
       "      <td>0.598585</td>\n",
       "      <td>-0.191285</td>\n",
       "      <td>0.566975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NP_958785</td>\n",
       "      <td>NaN</td>\n",
       "      <td>plectin isoform 1g</td>\n",
       "      <td>1.111370</td>\n",
       "      <td>2.650422</td>\n",
       "      <td>-0.648742</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.503899</td>\n",
       "      <td>2.779709</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.938210</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.055893</td>\n",
       "      <td>0.658143</td>\n",
       "      <td>-0.547749</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.606697</td>\n",
       "      <td>-0.183918</td>\n",
       "      <td>0.578702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NP_958786</td>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin isoform 1a</td>\n",
       "      <td>1.111370</td>\n",
       "      <td>2.650422</td>\n",
       "      <td>-0.654285</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.500619</td>\n",
       "      <td>2.779709</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.943919</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.065838</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.603993</td>\n",
       "      <td>-0.186022</td>\n",
       "      <td>0.576747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NP_000436</td>\n",
       "      <td>NaN</td>\n",
       "      <td>plectin isoform 1c</td>\n",
       "      <td>1.107561</td>\n",
       "      <td>2.646374</td>\n",
       "      <td>-0.632113</td>\n",
       "      <td>0.205377</td>\n",
       "      <td>-0.510459</td>\n",
       "      <td>2.797995</td>\n",
       "      <td>0.866423</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.935355</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.055893</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.603993</td>\n",
       "      <td>-0.186022</td>\n",
       "      <td>0.576747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NP_958781</td>\n",
       "      <td>NaN</td>\n",
       "      <td>plectin isoform 1e</td>\n",
       "      <td>1.115180</td>\n",
       "      <td>2.646374</td>\n",
       "      <td>-0.640428</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.503899</td>\n",
       "      <td>2.787023</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.935355</td>\n",
       "      <td>-0.503853</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.062523</td>\n",
       "      <td>0.651264</td>\n",
       "      <td>-0.556675</td>\n",
       "      <td>-0.395581</td>\n",
       "      <td>0.603993</td>\n",
       "      <td>-0.167079</td>\n",
       "      <td>0.576747</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 86 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  refseq_accession_number gene_symbol           gene_name  ao_a12d_01tcga   \n",
       "0               NP_958782        PLEC   plectin isoform 1        1.096131  \\\n",
       "1               NP_958785         NaN  plectin isoform 1g        1.111370   \n",
       "2               NP_958786        PLEC  plectin isoform 1a        1.111370   \n",
       "3               NP_000436         NaN  plectin isoform 1c        1.107561   \n",
       "4               NP_958781         NaN  plectin isoform 1e        1.115180   \n",
       "\n",
       "   c8_a131_01tcga  ao_a12b_01tcga  bh_a18q_02tcga  c8_a130_02tcga   \n",
       "0        2.609943       -0.659828        0.195341       -0.494060  \\\n",
       "1        2.650422       -0.648742        0.215413       -0.503899   \n",
       "2        2.650422       -0.654285        0.215413       -0.500619   \n",
       "3        2.646374       -0.632113        0.205377       -0.510459   \n",
       "4        2.646374       -0.640428        0.215413       -0.503899   \n",
       "\n",
       "   c8_a138_03tcga  e2_a154_03tcga  ...  ao_a12b_34tcga  a2_a0sw_35tcga   \n",
       "0        2.765081        0.862659  ...       -0.963904       -0.487772  \\\n",
       "1        2.779709        0.870186  ...       -0.938210       -0.487772   \n",
       "2        2.779709        0.870186  ...       -0.943919       -0.487772   \n",
       "3        2.797995        0.866423  ...       -0.935355       -0.487772   \n",
       "4        2.787023        0.870186  ...       -0.935355       -0.503853   \n",
       "\n",
       "   ao_a0jl_35tcga  bh_a0bv_35tcga  a2_a0ym_36tcga  bh_a0c7_36tcga   \n",
       "0        -0.10668       -0.065838        0.655850       -0.552212  \\\n",
       "1        -0.10668       -0.055893        0.658143       -0.547749   \n",
       "2        -0.10668       -0.065838        0.655850       -0.552212   \n",
       "3        -0.10668       -0.055893        0.655850       -0.552212   \n",
       "4        -0.10668       -0.062523        0.651264       -0.556675   \n",
       "\n",
       "   a2_a0sx_36tcga  263d3f_i_cptac  blcdb9_i_cptac  c4155b_c_cptac  \n",
       "0       -0.398560        0.598585       -0.191285        0.566975  \n",
       "1       -0.392601        0.606697       -0.183918        0.578702  \n",
       "2       -0.392601        0.603993       -0.186022        0.576747  \n",
       "3       -0.392601        0.603993       -0.186022        0.576747  \n",
       "4       -0.395581        0.603993       -0.167079        0.576747  \n",
       "\n",
       "[5 rows x 86 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_77proteomes.to_csv(\"./app/data/77_cancer_proteomes_CPTAC_itraq_with_columns.csv\", index=False)\n",
    "read_df_77proteomes = pd.read_csv(\"./app/data/77_cancer_proteomes_CPTAC_itraq_with_columns.csv\")\n",
    "read_df_77proteomes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
